sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -962,9 +962,8 @@ private String parentResourceLinkToQueryLink(String parentResourceLink, Resource
     private Mono<RxDocumentServiceResponse> delete(RxDocumentServiceRequest request, DocumentClientRetryPolicy documentClientRetryPolicy) {
         return populateHeaders(request, RequestVerb.DELETE)
             .flatMap(requestPopulated -> {
-                if (requestPopulated.requestContext != null && documentClientRetryPolicy.getRetryCount() > 0) {
-                    documentClientRetryPolicy.updateEndTime();
-                    requestPopulated.requestContext.updateRetryContext(documentClientRetryPolicy, true);
+                if (documentClientRetryPolicy.getRetryContext() != null && documentClientRetryPolicy.getRetryContext().getRetryCount() > 0) {
+                    documentClientRetryPolicy.getRetryContext().updateEndTime();
```

Comment:
Same code is being used multiple times below, you may want to extract it into a method?

Reply:
Its existing code, dont want to do the formatting , this is fine

Reply:
I disagree - fixing this would be good - especially because the current using pattern implicitly assumes IRetryPolicy.getRetryContext would never return null after initially returning a non-null value. Currently this is the case - but the interface doesn't really make it clear - so refactoring the updateEndTime into a separate helper method and operating on a snapshot of RetryContext to guarantee no NPE would be good.

Reply:
We are having  a single updateEndTime() in a if condition, so i don't think this is necessary to have a helper method for this.